libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.cpp

```diff
@@ -118,60 +100,62 @@ void AP_BattMonitor_SMBus_Generic::timer()
         _state.last_time_micros = tnow;
     }
 
-    read_full_charge_capacity();
-
     // FIXME: Perform current integration if the remaining capacity can't be requested
     read_remaining_capacity();
 
     read_temp();
-
-    read_serial_number();
-
-    read_cycle_count();
 }
 
-// read_block - returns number of characters read if successful, zero if unsuccessful
-uint8_t AP_BattMonitor_SMBus_Generic::read_block(uint8_t reg, uint8_t* data, bool append_zero) const
+// Read battery information messages  PEC, cycles, serial number, device name, manufacturer name, etc
+// Only returns false if pec check is not confirmed yet
+bool AP_BattMonitor_SMBus_Generic::initialize(uint32_t tnow)
 {
-    // get length
-    uint8_t bufflen;
-    // read byte (first byte indicates the number of bytes in the block)
-    if (!_dev->read_registers(reg, &bufflen, 1)) {
-        return 0;
+    if(_initialized){
+        return true;
     }
 
-    // sanity check length returned by smbus
-    if (bufflen == 0 || bufflen > SMBUS_READ_BLOCK_MAXIMUM_TRANSFER) {
-        return 0;
+    // always first check if PEC is supported
+    if (!check_pec_support()) {
+        return false;
     }
 
-    // buffer to hold results (2 extra byte returned holding length and PEC)
-    const uint8_t read_size = bufflen + 1 + (_pec_supported ? 1 : 0);
-    uint8_t buff[read_size];
+    // check cell count
+    if (_state.healthy) {
```

Comment:
Unless I've missed something healthy isn't actual true initially, and it's normally toggled by the polling loop. Why are we looking at it at all inside of the initialization block?

Especially since this is called repeatedly during polling until the initialized flag (which can only be set by entering here) doesn't this mean we might repeatedly poll for all the smart battery info for ever on a non healthy battery and massively bring down the I2C bus by accident?